hera_cal/datacontainer.py

```diff
@@ -558,14 +573,22 @@ def _add_red(self, ubl_key, red):
             self._red_key_to_bls[ubl_key].append(bl)
             self._red_key_to_bls[reverse_bl(ubl_key)].append(reverse_bl(bl))
 
-    def build_red_keys(self, reds):
+    def build_red_keys(self, reds: RedundantGroups | list[list[Baseline]]):
         '''Build the dictionaries that map baselines to redundant keys.
 
         Arguments:
             reds: list of lists of redundant baseline tuples, e.g. (ind1, ind2, pol).
         '''
         # Map all redundant keys to the same underlying data
+        self._red_key_to_bls = {}
+        self._bl_to_red_key = {}
+        
         self.reds = []
+        if isinstance(reds, RedundantGroups):
+            self.red_groups = reds
+        else:
+            self.red_groups = RedundantGroups(red_list=reds, antpos=getattr(self, 'antpos', None))
```

Comment:
If it's not a `RedundantGroups` object, we should recast it on __init__

Reply:
I'm recasting it there, but of course you can also call `build_red_keys` directly so we need to also cast it here.